### DISCUSSION TITLE
lots of "Library item not found" and "Library item doesn't exist undefined" log errors

### DISCUSSION THREAD

**Discussion by mtklr** [advplyr/audiobookshelf#3091] [Q&A]
I hesitate to ask because everything is working fine, just after importing a few titles I start seeing these repeated errors in the logs and I'm wondering what may be causing it.

```
...
[2024-06-20 17:49:41.374] ERROR: [MeController] batchUpdateMediaProgress: Library Item does not exist undefined (MeController.js:150)
[2024-06-20 17:49:41.375] ERROR: [LibraryItem] Library item not found with id "a4092b77-ce5d-47bf-a6ed-7352916d06fe" (LibraryItem.js:496)
[2024-06-20 17:49:41.376] ERROR: [MeController] batchUpdateMediaProgress: Library Item does not exist undefined (MeController.js:150)
[2024-06-20 17:49:41.377] ERROR: [LibraryItem] Library item not found with id "333b470d-1708-46d8-8cfb-f4ca5548a133" (LibraryItem.js:496)
[2024-06-20 17:49:41.377] ERROR: [MeController] batchUpdateMediaProgress: Library Item does not exist undefined (MeController.js:150)
...
```

Running in docker on debian... again, more curious than anything it's been really nice using this for audiobooks.

Thanks,
mk

**Comment by jhoff80** [comment]
I have the same issue.  Proxmox / LXC version of audiobookshelf.

In my case, I added a book that a user started listening to in Plappa, then reorganized my folder structure and so some of the original book entries got broken (and were deleted on the server) and new entries were created for the same books.  I suspect their version of the Plappa app is still trying to look for that deleted item, but I don't know for sure.

**Reply by AncientMystic** [reply]
I am having the same issue, also using proxmox/LXC, the server either fails to find items or disappears completely for a few minutes then its back and the items are back then it does it again, like its just going up and down in the background and "missing" items are there and it finds them after 5-20 min and plays the content fine usually, its only dropped out and said it couldn't find it again a few times. 

But either way its like part of audiobookshelf is cycling up and down in the background or something.

**Reply by nichwall** [reply]
ABS disappearing and reappearing like that is usually because your server is crashing and auto restarting. You'll want to look at the logs.

**Reply by AncientMystic** [reply]
> ABS disappearing and reappearing like that is usually because your server is crashing and auto restarting. You'll want to look at the logs.

Thank you for your suggestion. 

I found the fix, in my case at least, it was my ZFS drive was a bit overwhelmed and the scans and other actions of audiobookshelf among all the other use was causing it to lock up for a second here and there to flush everything to disk, so i moved audiobookshelfs storage to an SSD and now it is completely fine. 

So i guess on proxmox, don't host the LXC off ZFS unless you have a large raid. 

Now my only issue is audiobookshelf is only detecting about 27% of my library.
